localstack-core/localstack/services/lambda_/event_source_mapping/pollers/stream_poller.py

```diff
@@ -187,9 +210,26 @@ def poll_events_from_shard(self, shard_id: str, shard_iterator: str):
         # TODO: think about how to avoid starvation of other shards if one shard runs into infinite retries
         attempts = 0
         error_payload = {}
-        while not abort_condition and not self.max_retries_exceeded(attempts):
+
+        boff = ExponentialBackoff(max_retries=attempts)
+        while (
+            not abort_condition
+            and not self.max_retries_exceeded(attempts)
+            and not self._is_shutdown.is_set()
+        ):
             try:
+                if self.has_record_expired(polled_events[-1]):
+                    abort_condition = "RecordAgeExpired"
+
+                if attempts > 0:
+                    # TODO: Should we always backoff (with jitter) before processing since we may not want multiple pollers
```

Comment:
Starting with backoff after a retry seems a good solution to start with for now, not impacting any best-case responsiveness